.ci/docker/build-and-publish-docker-image.sh

```diff
@@ -14,8 +14,8 @@ elif [[ "$1" != "" ]]; then
   exit 1
 fi
 
-GHC_VERSIONS=(  "9.4.3"  "9.2.5"   "9.0.2"   "8.10.7"  "8.8.4"   "8.6.5")
-CABAL_VERSIONS=("3.8.1.0" "3.6.2.0" "3.4.0.0" "3.2.0.0" "3.2.0.0" "3.0.0.0")
+GHC_VERSIONS=(  "9.6.1"  "9.4.3"  "9.2.5"   "9.0.2"   "8.10.7"  "8.8.4"   "8.6.5")
+CABAL_VERSIONS=("3.10.1.0" "3.8.1.0" "3.6.2.0" "3.4.0.0" "3.2.0.0" "3.2.0.0" "3.0.0.0")
```

Comment:
This is getting quite a lot to test. I think we should only run tests on the last three versions for PRs, and run the full suite at night. (We should build everything on PRs though.) What do you think?

Reply:
I agree, but perhaps make that change on a separate PR.

Reply:
Yeah, totally